src/components/Header/NameExiticon.jsx

```diff
@@ -0,0 +1,38 @@
+import React from "react"
+import { useSelector, useDispatch } from "react-redux"
+import { NavLink } from 'react-router-dom'
+
+import { fetchLogout } from "../../ducks/userInfo"
+import {
+    ExitIconWrapper,
+    MenuInitiaLetterCircle,
+    MenuInitiaLetter,
+    MenuName
+} from "./Header.styled"
+import ExitIcon from "../../assets/image/Header-exit-icon.svg"
+
+const NameExiticon = () => {
+    let userName = 'email'
+    const dispatch = useDispatch()
+    const logout = () => dispatch(fetchLogout(formData))
+    const userData = useSelector(state => state.userInfo.userData)
+    if (userData) {
+        userName = userData.email
+    }
+    const nameLetter = userName && (userName.charAt(0))
```

Comment:
Copy&paste from component above, maybe you need to pass this data from some parent component or separate them in another way

Reply:
All redone as you said. I removed these
(repeating) blocks from two files (MobileMenu, Logouticon). And instead included this block in the HeaderUser parent. From where, with the help of props, I passed all the necessary properties ( props.userName, props.nameLetter, and the logoutProps function).
Thank you for the lesson))

![image](https://user-images.githubusercontent.com/96341034/189454058-5bef23d7-deb1-430f-bc8d-131075e0409b.png)